src/com/hms/flexyosisoftconnector/OSIsoftConfig.java

```diff
@@ -146,9 +185,98 @@ public static void initConfig(String configFile) throws JSONException, IOExcepti
             + "&Content-Type=application/json"
             + "&X-Requested-With=JSONHttpRequest"
             + "&messageformat=json&omfversion=1.1";
+    ocsPostHeaders =
+        "Authorization=Bearer "
+            + OSIsoftConfig.getOCSToken()
+            + "&Content-Type=application/json"
+            + "&messageformat=json&omfversion=1.1"
+            + "&Producertoken="
+            + OSIsoftConfig.getOCSToken();
+
+    Logger.LOG_DEBUG(ocsPostHeaders);
     setFlexyName();
   }
 
+  /**
+   * The token will be retrieved in a BASIC script due to issues with the request when made in Java.
+   * Assume the response file exists and read from it.
+   */
+  private static void setOCSToken() {
+    // file name hardcoded in basic script and should never change.
+    String tokenFile = "/usr/token.json";
+    JSONTokener JsonT = null;
+    String token = "";
+    try {
+      JsonT = new JSONTokener(FileAccessManager.readFileToString(tokenFile));
+      // Build a JSON Object containing the whole file
+      JSONObject tokenJSON = new JSONObject(JsonT);
+
+      token = tokenJSON.getString("access_token");
+    } catch (IOException e) {
+      Logger.LOG_SERIOUS("Unable to read OCS token from https request response file.");
+      Logger.LOG_EXCEPTION(e);
+    } catch (JSONException e) {
+      Logger.LOG_SERIOUS(
+          "Unable to parse JSON in OCS token response file from OCS new token https request.");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Set the access token
+    OcsOmfToken = token;
+  }
+
+  /**
+   * This function need to have a basic script running that is specified in the README. Updates the
+   * OCS token if the OCS token tag is not set to 0. This tag will then be reset to 0.
+   */
+  public static void updateOcsToken() {

```

Comment:
setOCSToken() and updateOcsToken() have different capitalization for OCS

Reply:
Since OCS is three characters it should be Ocs for all usage